File: apps/web/src/types/activity-log.ts
/**
 * ActivityEventType (spec/api.md#List Activity Logs).
 * MANUAL_NOTE is historical only — new rows are system-generated.
 */
export type ActivityEventType =
  | "MANUAL_NOTE"
  | "PROGRESS_UPDATED"
  | "COMPANY_STATUS_UPDATED"
  | "PHASE_UPDATED"
  | "ASSIGNMENT_UPDATED"
  | "CONTRACT_CREATED"
  | "CONTRACT_MENU_STATUS_UPDATED"
  | "PAYMENT_STATUS_UPDATED"

export type ActivityLog = {
  id: string
  yearlyCompanyId: string
  eventType: ActivityEventType
  message: string
  createdAt: string
  createdById: string | null
  createdByName: string | null
}

export const ACTIVITY_EVENT_LABEL: Record<ActivityEventType, string> = {
  MANUAL_NOTE: "手動メモ",
  PROGRESS_UPDATED: "進捗更新",
  COMPANY_STATUS_UPDATED: "企業ステータス",
  PHASE_UPDATED: "フェーズ",
  ASSIGNMENT_UPDATED: "担当変更",
  CONTRACT_CREATED: "契約作成",
  CONTRACT_MENU_STATUS_UPDATED: "協賛メニュー",
  PAYMENT_STATUS_UPDATED: "入金",
}

export function activityEventLabel(eventType: string): string {
  return ACTIVITY_EVENT_LABEL[eventType as ActivityEventType] ?? eventType
}

File: apps/api/internal/model/activity_log.go
package model

// ActivityEventType values returned as eventType (spec/api.md#List Activity Logs).
// Persistence still stores them in activity_logs.action.
const (
	EventProgressUpdated           = "PROGRESS_UPDATED"
	EventCompanyStatusUpdated      = "COMPANY_STATUS_UPDATED"
	EventPhaseUpdated              = "PHASE_UPDATED"
	EventAssignmentUpdated         = "ASSIGNMENT_UPDATED"
	EventContractCreated           = "CONTRACT_CREATED"
	EventContractMenuStatusUpdated = "CONTRACT_MENU_STATUS_UPDATED"
	EventPaymentStatusUpdated      = "PAYMENT_STATUS_UPDATED"
	EventManualNote                = "MANUAL_NOTE"
)

// NormalizeEventType maps legacy action values written before spec/api.md
// alignment onto the current eventType set. Unknown values are returned as-is
// so a badge can still show the raw token.
func NormalizeEventType(action string) string {
	switch action {
	case "UPDATE_PROGRESS":
		return EventProgressUpdated
	case "UPDATE_COMPANY_STATUS":
		return EventCompanyStatusUpdated
	case "UPDATE_PHASE":
		return EventPhaseUpdated
	case "ASSIGNED_MEMBER", "CLEARED_MEMBER":
		return EventAssignmentUpdated
	case "CONTRACT_MENU_SUBMITTED":
		return EventContractMenuStatusUpdated
	case "PAYMENT_CONFIRMED":
		return EventPaymentStatusUpdated
	default:
		return action
	}
}
